### PR TITLE
Commit #3.

### PR DESCRIPTION
**Pull request by JuanMiChee**
<H3>Add new card to the list feature:</H3>
Now the entire implementation of adding a new card to the list is done, including its storage, and management with view and viewModel correctly, i added a view for this action, where we can write the name of the new cell we want to add.

<H3>- I also corrected some error just as: -</H3>

1. Using the actual object, instead of its abstraction at *viewModel.dependencies*

2. Implementation of build function for important views

3. Moving annoying styles to its very own folder, so we can use them without repeating them every single time we need (Buttons for example).

  4.Dividing huge views, into segments, easier to read

<H1>Evidence:</H1>

### CODE REVIEW

**Comment by JuanMiChee** [comment]
This Pr is looking good, well documented, and well written. There is a detail i dont really like tho, you are uploading view changes, with bussines and storage changes, try to split your changes better next time. Something like this:

1. Create AddNewCardFeatureBranch
2. Create another branch from it, called ViewModelChances, so you can make all of your changes in there, so it can be checked by your team easier, and not make us read the whole thing, just makes us read "Hey, this this and that uploaded at viewModel"
3. Continue creating branches like this, one for view changes, one for UI Correcting, another for storage, etc etc, this way will be a lot easier to read for everybody, and your work will be very well organized.
4. once you finish you work with a branch, make a PR to the main one "AddNewCardFeatureBranch" in this case. and once you finish all of the branches and work on them, you can make the PR to develop

I hope you take this as a recomendation and not as a harassment, you bozo. Greetings <3

File: FlashEnglishCards/Storage/MainStorage.swift [hunk 1]
```diff
@@ -7,6 +7,38 @@
...
+struct MainStorage: StorageProtocol {
+    let defaults = UserDefaults.standard
+    
+    func saveNewCard(card: FlashCardModel) {
+        var arrayToSave: [FlashCardModel] = getCards()
+        arrayToSave.append(card)
+        
+        do {
+            let encoder = JSONEncoder()
+            if let encodedFlashCardArray = try? encoder.encode(arrayToSave) {
```
Comment (angelbetancourt) [inline FlashEnglishCards/Storage/MainStorage.swift:24]: Make this function throwable and just try here

File: FlashEnglishCards/Storage/MainStorage.swift [hunk 2]
```diff
@@ -7,6 +7,38 @@
...
+                print(encodedFlashCardArray)
+                defaults.set(encodedFlashCardArray, forKey: "flashCardsArray")
```
Comment (angelbetancourt) [inline FlashEnglishCards/Storage/MainStorage.swift:26]:
```suggestion
                defaults.set(encodedFlashCardArray, forKey: .cards)
```

File: FlashEnglishCards/ViewBuilders/HomeViewBuilder.swift [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+//
+//  Created by Juan Harrington on 10/11/23.
+//
+
+import Foundation
+
+extension HomeView {
+    static func build() -> Self {
+        let storage = MainStorage()
+        return HomeView(viewModel: HomeViewViewModel(dependencies: HomeViewViewModel.Dependencies(getCards: GetCardsUseCase(storage: storage), saveCards: SaveNewCardUseCase(storage: storage))))
```
Comment (angelbetancourt) [inline FlashEnglishCards/ViewBuilders/HomeViewBuilder.swift]:
```suggestion
       .init(viewModel: .init(dependencies: .init(getCards: getCards, saveCards: saveCargs)))
```

File: FlashEnglishCards/Views/HomeView.swift [hunk 4]
```diff
@@ -7,42 +7,84 @@
 
 import SwiftUI
 
-
-
 struct HomeView: View {
     
-    @ObservedObject var viewModel = HomeViewViewModel(dependencies: HomeViewViewModel.Dependencies(getCards: GetCardsUseCase()))
+    @ObservedObject var viewModel: HomeViewViewModel
```
Comment (angelbetancourt) [inline FlashEnglishCards/Views/HomeView.swift]:
```suggestion
    @StateObject var viewModel: HomeViewViewModel
```

File: FlashEnglishCards/Views/HomeView.swift [hunk 5]
```diff
@@ -7,42 +7,84 @@
...
-                .font(.title3)
-                .foregroundColor(.gray)
-            ScrollView {
-                ForEach(viewModel.viewContent.texts) { text in
-                    FlashCardView(text: text.text)
+            Spacer()
+            HStack {
+                Spacer()
+                Button("+") {
+                    viewModel.isSheetPresented = true
```
Comment (angelbetancourt) [inline FlashEnglishCards/Views/HomeView.swift]:
```suggestion
                    viewModel.handleAdd()
```

File: FlashEnglishCards/Views/HomeView.swift [hunk 6]
```diff
@@ -7,42 +7,84 @@
...
+    
+    //Here is the sheet view
+    var sheetView: some View {
+        VStack {
+            Text("*Name of your new card.*")
+                .font(.title)
+                .foregroundColor(Color(UIColor.systemGray))
+            TextField(
+                "Card name",
+                text: $viewModel.sheetText
```
Comment (angelbetancourt) [inline FlashEnglishCards/Views/HomeView.swift]:
```suggestion
                text: $viewModel.newCardTitle
```

**Review by angelbetancourt — CHANGES_REQUESTED** [review]
Just some minot comments

**Pull request merged**